diff --git a/build/soong/rust/bindgen.go b/build/soong/rust/bindgen.go
index 2f84168bd..bc26590fa 100644
--- a/build/soong/rust/bindgen.go
+++ b/build/soong/rust/bindgen.go
@@ -29,7 +29,7 @@ var (
 	defaultBindgenFlags = []string{""}
 
 	// bindgen should specify its own Clang revision so updating Clang isn't potentially blocked on bindgen failures.
-	bindgenClangVersion = "clang-r530567"
+	bindgenClangVersion = "clang-r547379"
 
 	_ = pctx.VariableFunc("bindgenClangVersion", func(ctx android.PackageVarContext) string {
 		if override := ctx.Config().Getenv("LLVM_BINDGEN_PREBUILTS_VERSION"); override != "" {
