diff --git a/external/tensorflow/tensorflow/lite/kernels/elementwise.cc b/external/tensorflow/tensorflow/lite/kernels/elementwise.cc
index 4dbd5f2ae72..f749b3a28db 100644
--- a/external/tensorflow/tensorflow/lite/kernels/elementwise.cc
+++ b/external/tensorflow/tensorflow/lite/kernels/elementwise.cc
@@ -227,7 +227,7 @@ TfLiteStatus AbsEval(TfLiteContext* context, TfLiteNode* node) {
   const TfLiteType type = input->type;
   switch (type) {
     case kTfLiteFloat32:
-      return EvalImpl<float>(context, node, std::abs<float>, type);
+      return EvalImpl<float>(context, node, [](float f) { return std::abs(f); }, type);
     case kTfLiteInt8:
       return AbsEvalQuantized<int8_t>(context, node, type);
     case kTfLiteInt16:
