diff --git a/prebuilts/clang/host/linux-x86/soong/clangprebuilts.go b/prebuilts/clang/host/linux-x86/soong/clangprebuilts.go
index 3406023e..8ee03d0f 100644
--- a/prebuilts/clang/host/linux-x86/soong/clangprebuilts.go
+++ b/prebuilts/clang/host/linux-x86/soong/clangprebuilts.go
@@ -35,7 +35,6 @@ var (
 		"bin/llvm-objcopy",
 		"bin/llvm-strip",
 		"bin/llvm-symbolizer",
-		"lib/libc++.so",
 		"lib/x86_64-unknown-linux-gnu/libc++.so",
 	}
 )
@@ -375,7 +374,6 @@ func llvmPrebuiltBuildTool(ctx android.LoadHookContext) {
 	name := strings.TrimPrefix(ctx.ModuleName(), "prebuilt_")
 	src := path.Join(clangDir, "bin", name)
 	deps := []string{
-		path.Join(clangDir, "lib", "libc++.so"),
 		path.Join(clangDir, "lib", "x86_64-unknown-linux-gnu", "libc++.so"),
 	}
 
@@ -462,10 +460,10 @@ func libClangRtPrebuiltLibraryShared(ctx android.LoadHookContext, libProps *preb
 	p.Target.Linux_bionic_arm64.Stem = proptools.StringPtr(name + "-aarch64-android" + suffix)
 	p.Target.Linux_bionic_x86_64.Srcs = []string{path.Join(libDir, name+"-x86_64-android"+suffix+".so")}
 	p.Target.Linux_bionic_x86_64.Stem = proptools.StringPtr(name + "-x86_64-android" + suffix)
-	p.Target.Glibc_x86.Srcs = []string{path.Join(libDir, name+"-i386"+suffix+".so")}
-	p.Target.Glibc_x86.Stem = proptools.StringPtr(name + "-i386" + suffix)
-	p.Target.Glibc_x86_64.Srcs = []string{path.Join(libDir, name+"-x86_64"+suffix+".so")}
-	p.Target.Glibc_x86_64.Stem = proptools.StringPtr(name + "-x86_64" + suffix)
+	p.Target.Glibc_x86.Srcs = []string{path.Join(libDir, "../i386-unknown-linux-gnu", name+suffix+".so")}
+	p.Target.Glibc_x86.Stem = proptools.StringPtr(name + suffix)
+	p.Target.Glibc_x86_64.Srcs = []string{path.Join(libDir, "../x86_64-unknown-linux-gnu", name+suffix+".so")}
+	p.Target.Glibc_x86_64.Stem = proptools.StringPtr(name + suffix)
 	p.Target.Linux_musl_x86.Srcs = []string{path.Join(libDir, "i686-unknown-linux-musl/lib/linux", name+"-i386"+suffix+".so")}
 	p.Target.Linux_musl_x86.Stem = proptools.StringPtr(name + "-i386" + suffix)
 	p.Target.Linux_musl_x86_64.Srcs = []string{path.Join(libDir, "x86_64-unknown-linux-musl/lib/linux", name+"-x86_64"+suffix+".so")}
@@ -530,8 +528,8 @@ func libClangRtPrebuiltLibraryStatic(ctx android.LoadHookContext, libProps *preb
 	p.Target.Android_x86_64.Srcs = []string{path.Join(libDir, name+"-x86_64-android"+suffix+".a")}
 	p.Target.Linux_bionic_arm64.Srcs = []string{path.Join(libDir, name+"-aarch64-android"+suffix+".a")}
 	p.Target.Linux_bionic_x86_64.Srcs = []string{path.Join(libDir, name+"-x86_64-android"+suffix+".a")}
-	p.Target.Glibc_x86.Srcs = []string{path.Join(libDir, name+"-i386"+suffix+".a")}
-	p.Target.Glibc_x86_64.Srcs = []string{path.Join(libDir, name+"-x86_64"+suffix+".a")}
+	p.Target.Glibc_x86.Srcs = []string{path.Join(libDir, "../i386-unknown-linux-gnu", name+suffix+".a")}
+	p.Target.Glibc_x86_64.Srcs = []string{path.Join(libDir, "../x86_64-unknown-linux-gnu", name+suffix+".a")}
 	p.Target.Linux_musl_x86.Srcs = []string{path.Join(libDir, "i686-unknown-linux-musl/lib/linux", name+"-i386"+suffix+".a")}
 	p.Target.Linux_musl_x86_64.Srcs = []string{path.Join(libDir, "x86_64-unknown-linux-musl/lib/linux", name+"-x86_64"+suffix+".a")}
 	p.Target.Linux_musl_arm.Srcs = []string{path.Join(libDir, "arm-unknown-linux-musleabihf/lib/linux", name+"-armhf"+suffix+".a")}
@@ -567,8 +565,8 @@ func libClangRtPrebuiltObject(ctx android.LoadHookContext) {
 	name := strings.TrimPrefix(ctx.ModuleName(), "prebuilt_")
 
 	p := &props{}
-	p.Arch.X86.Srcs = []string{path.Join(libDir, name+"-i386.o")}
-	p.Arch.X86_64.Srcs = []string{path.Join(libDir, name+"-x86_64.o")}
+	p.Arch.X86.Srcs = []string{path.Join(libDir, "../i386-unknown-linux-gnu", name+".o")}
+	p.Arch.X86_64.Srcs = []string{path.Join(libDir, "../x86_64-unknown-linux-gnu", name+".o")}
 	p.Arch.Arm.Srcs = []string{path.Join(libDir, "arm-unknown-linux-musleabihf/lib/linux", name+"-armhf.o")}
 	p.Arch.Arm64.Srcs = []string{path.Join(libDir, "aarch64-unknown-linux-musl/lib/linux", name+"-aarch64.o")}
 	p.System_shared_libs = []string{}
