diff --git a/frameworks/base/media/jni/android_media_MediaMetricsJNI.cpp b/frameworks/base/media/jni/android_media_MediaMetricsJNI.cpp
index 2769dbc..0d2ad85 100644
--- a/frameworks/base/media/jni/android_media_MediaMetricsJNI.cpp
+++ b/frameworks/base/media/jni/android_media_MediaMetricsJNI.cpp
@@ -222,7 +222,7 @@
 
 // ----------------------------------------------------------------------------
 
-static constexpr JNINativeMethod gMethods[] = {
+static const JNINativeMethod gMethods[] = {
     {"native_submit_bytebuffer", "(Ljava/nio/ByteBuffer;I)I",
             (void *)android_media_MediaMetrics_submit_bytebuffer},
 };
