diff --git a/packages/modules/Virtualization/guest/trusty/security_vm/vm/Android.bp b/packages/modules/Virtualization/guest/trusty/security_vm/vm/Android.bp
index 77ac95407..d85e3a5f7 100644
--- a/packages/modules/Virtualization/guest/trusty/security_vm/vm/Android.bp
+++ b/packages/modules/Virtualization/guest/trusty/security_vm/vm/Android.bp
@@ -13,8 +13,8 @@ package {
 prebuilt_etc {
     name: "trusty_security_vm.elf",
     src: select((os(), arch(), soong_config_variable("trusty_system_vm", "enabled")), {
-        ("android", "arm64", true): ":trusty_security_vm_signed",
-        ("android", "x86_64", true): ":trusty_security_vm_unsigned",
+        // ("android", "arm64", true): ":trusty_security_vm_signed",
+        // ("android", "x86_64", true): ":trusty_security_vm_unsigned",
         (default, default, default): ":empty_file",
     }),
     system_ext_specific: true,
