diff --git a/external/llvm/utils/TableGen/AsmWriterEmitter.cpp b/external/llvm/utils/TableGen/AsmWriterEmitter.cpp
index fc2138f7e8..ef08283e17 100644
--- a/external/llvm/utils/TableGen/AsmWriterEmitter.cpp
+++ b/external/llvm/utils/TableGen/AsmWriterEmitter.cpp
@@ -727,7 +727,7 @@ static unsigned CountNumOperands(StringRef AsmString, unsigned Variant) {
 namespace {
 struct AliasPriorityComparator {
   typedef std::pair<CodeGenInstAlias, int> ValueType;
-  bool operator()(const ValueType &LHS, const ValueType &RHS) {
+  bool operator()(const ValueType &LHS, const ValueType &RHS) const {
     if (LHS.second ==  RHS.second) {
       // We don't actually care about the order, but for consistency it
       // shouldn't depend on pointer comparisons.

diff --git a/external/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp b/external/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp
index 8a3c4d14fe..6cb820f665 100644
--- a/external/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp
+++ b/external/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp
@@ -3145,7 +3145,7 @@ void BoUpSLP::scheduleBlock(BlockScheduling *BS) {
   // sorted by the original instruction location. This lets the final schedule
   // be as  close as possible to the original instruction order.
   struct ScheduleDataCompare {
-    bool operator()(ScheduleData *SD1, ScheduleData *SD2) {
+    bool operator()(ScheduleData *SD1, ScheduleData *SD2) const {
       return SD2->SchedulingPriority < SD1->SchedulingPriority;
     }
   };
