diff --git a/system/libbase/properties.cpp b/system/libbase/properties.cpp
index e217c9e..9fd44a8 100644
--- a/system/libbase/properties.cpp
+++ b/system/libbase/properties.cpp
@@ -60,9 +60,9 @@ struct prop_info {
 
 struct prop_info_cmp {
   using is_transparent = void;
-  bool operator()(const prop_info& lhs, const prop_info& rhs) { return lhs.key < rhs.key; }
-  bool operator()(std::string_view lhs, const prop_info& rhs) { return lhs < rhs.key; }
-  bool operator()(const prop_info& lhs, std::string_view rhs) { return lhs.key < rhs; }
+  bool operator()(const prop_info& lhs, const prop_info& rhs) const { return lhs.key < rhs.key; }
+  bool operator()(std::string_view lhs, const prop_info& rhs) const { return lhs < rhs.key; }
+  bool operator()(const prop_info& lhs, std::string_view rhs) const { return lhs.key < rhs; }
 };
 
 static auto& g_properties_lock = *new std::mutex;
