diff --git a/external/v4l-utils/Android.bp b/external/v4l-utils/Android.bp
index 999e004f..712baaff 100644
--- a/external/v4l-utils/Android.bp
+++ b/external/v4l-utils/Android.bp
@@ -32,6 +32,11 @@ cc_defaults {
     cflags: [
         "-DPACKAGE_VERSION=\"" + v4l_utils_version + "\"",
         "-DNO_LIBV4L2",
+        // _LIBCPP_FALLTHROUGH() is not available in Clang
+        // anymore so this is a temporary fix for it.
+        // Issue raised to Linux Media mailing list for a fix:
+        // https://lore.kernel.org/linux-media/CALPBKXJXg4R4jm_RZ2QDnQuOf_hGwJRQ8L=1H-nthuDHaJMq+Q@mail.gmail.com
+        "-D_LIBCPP_FALLTHROUGH()=[[fallthrough]]",
 
         "-Wno-shift-overflow",
     ],
