diff --git a/services/inputflinger/reader/mapper/TouchInputMapper.cpp b/services/inputflinger/reader/mapper/TouchInputMapper.cpp
index a6dea4e..c4a1008 100644
--- a/services/inputflinger/reader/mapper/TouchInputMapper.cpp
+++ b/services/inputflinger/reader/mapper/TouchInputMapper.cpp
@@ -83,13 +83,6 @@ static std::string toString(const InputDeviceUsiVersion& v) {
     return base::StringPrintf("%d.%d", v.majorVersion, v.minorVersion);
 }
 
-template <typename T>
-inline static void swap(T& a, T& b) {
-    T temp = a;
-    a = b;
-    b = temp;
-}
-
 inline static int32_t signExtendNybble(int32_t value) {
     return value >= 8 ? value - 16 : value;
 }
@@ -2438,7 +2431,7 @@ void TouchInputMapper::assignPointerIds(const RawState& last, RawState& current)
                 break;
             }
 
-            swap(heap[parentIndex], heap[childIndex]);
+            std::swap(heap[parentIndex], heap[childIndex]);
             parentIndex = childIndex;
         }
     }
@@ -2484,7 +2477,7 @@ void TouchInputMapper::assignPointerIds(const RawState& last, RawState& current)
                         break;
                     }
 
-                    swap(heap[parentIndex], heap[childIndex]);
+                    std::swap(heap[parentIndex], heap[childIndex]);
                     parentIndex = childIndex;
                 }
 
